src/vs/editor/contrib/referenceSearch/browser/referencesWidget.ts

```diff
@@ -212,8 +212,19 @@ class DataSource implements tree.IDataSource {
 	}
 }
 
+/**
+ * RevealedReference wraps an optional OneReference.
+ * This allows RevealedReference to be passed around (i.e. to Controller)
+ * and updates to the reference can be seen.
+ */
+type RevealedReference = { ref: OneReference | undefined };
```

Comment:
I suspect that you misread this to be a collection of references when it is just a container for a single reference.

this could have just been `type RevealedReference = OneReference | undefined;` except I needed to be able to pass this through to a Controller and have the controller see the updated OneReference if it changed. That is why I created a wrapper object with a single field `ref` to store the reference.

Does that make sense? Can you think of a way for me to make that more clear or is there a better way for me to accomplish this?